newrelic/hooks/mlmodel_langchain.py

```diff
@@ -256,6 +258,132 @@ def wrap_similarity_search(wrapped, instance, args, kwargs):
     return response
 
 
+def wrap_tool_sync_run(wrapped, instance, args, kwargs):
+    transaction = current_transaction()
+    if not transaction:
+        return wrapped(*args, **kwargs)
+
+    # Framework metric also used for entity tagging in the UI
+    transaction.add_ml_model_info("Langchain", LANGCHAIN_VERSION)
+
+    run_args = bind_args(wrapped, args, kwargs)
+
+    metadata = {}
+    metadata.update(run_args.get("metadata") or {})
+    metadata.update(getattr(instance, "metadata", None) or {})
+
+    tags = []
+    tags.extend(run_args.get("tags") or [])
+    tags.extend(getattr(instance, "tags", None) or [])
+
+    tool_input = run_args.get("tool_input", "")
+    tool_name = instance.name or ""
+    tool_description = instance.description or ""
+
+    span_id = None
+    trace_id = None
+
+    settings = transaction.settings if transaction.settings is not None else global_settings()
+    app_name = settings.app_name
+    tool_id = str(uuid.uuid4())
+
+    function_name = wrapped.__name__
+
+    with FunctionTrace(name=function_name, group="Llm/tool/Langchain") as ft:
+        # Get trace information
+        available_metadata = get_trace_linking_metadata()
+        span_id = available_metadata.get("span.id", "")
+        trace_id = available_metadata.get("trace.id", "")
+
+        try:
+            return_val = wrapped(**run_args)
+        except Exception as exc:
+            ft.notice_error(
+                attributes={
+                    "tool_id": tool_id,
+                }
+            )
+
+            run_manager_info = getattr(transaction, "_nr_run_manager_tools_info", {})
```

Comment:
Might as well just grab this whole thing in one line.
```suggestion
            run_id = getattr(transaction, "_nr_run_manager_tools_info", {}).get("run_id", "")
```